### DISCUSSION TITLE
Port communication security

### DISCUSSION THREAD

**Discussion by nermin-pezerovic** [rustdesk/rustdesk#5066] [General]
Hello, there are several ports that are used by rustdesk(21116,21117,21118,21119 i think). They are used to relay messages between server and clients. I am wondering how secure is this connection? I have rustdesk server with public IP address and it communicates over the internet with client machines.

**Comment by dinger1986** [comment]
are you using a key?

**Reply by nermin-pezerovic** [reply]
Yes, there is a key that is generated when installing rustdesk on a server.

**Reply by nermin-pezerovic** [reply]
I think that key is for authentication, right? Does it somehow encrypt the traffic, session?

**Reply by dinger1986** [reply]
ok, that encrypts the connection end to end. The encryption algorithm used by RustDesk is the ChaCha20-Poly1305 cipher suite, which is considered to be one of the strongest and most secure encryption algorithms available today.

**Reply by nermin-pezerovic** [reply]
Great, thank you very much!
